gapis/api/vulkan/errors.api

```diff
@@ -0,0 +1,83 @@
+// Copyright (C) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Based off of the original vulkan.h header file which has the following
+// license.
+
+// Copyright (c) 2015 The Khronos Group Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and/or associated documentation files (the
+// "Materials"), to deal in the Materials without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Materials, and to
+// permit persons to whom the Materials are furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Materials.
+//
+// THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+// CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+
+extern void onVkError(any err)
+
+sub void vkErrorInvalidInstance(VkInstance inst) {
+  vkErrorInvalidHandle("VkInstance", as!u64(inst))
+}
+
+sub void vkErrorInvalidDevice(VkDevice dev) {
+  vkErrorInvalidHandle("VkDevice", as!u64(dev))
+}
+
+sub void vkErrorInvalidImage(VkImage img) {
+  vkErrorInvalidHandle("VkImage", as!u64(img))
+}
+
+sub void vkErrorInvalidQueue(VkQueue queue) {
+  vkErrorInvalidHandle("VkQueue", as!u64(queue))
+}
+
+sub void vkErrorInvalidHandle(string handleType, u64 handle) {
+  onVkError(new!ERR_INVALID_HANDLE(handleType: handleType, handle: handle))
+  abort
+}
+
+sub void vkErrorNullPointer(string pointerType) {
+  onVkError(new!ERR_NULL_POINTER(pointerType: pointerType))
+  abort
+}
+
+sub void vkErrorUnsupportedExtension(const char* name) {
+  onVkError(new!ERR_UNSUPPORTED_EXTENSION(name: as!string(name)))
+  // Continue the mutation as this may not cause problem.
+}
+
+class ERR_INVALID_HANDLE {
```

Comment:
These should be generated by the stringtable system. Declare them in `en-us.stb.md` and `import "../../messages/messages.api"`

Reply:
Importing the `../../messages/message.api` causes an error on the CPP side:
```
.../gapid/gapii/cc/vulkan_types.h:1983:27: error: redefinition of 'SEVERITY_DEBUG'
```

`Severity` is defined in both `gles_types.h` and `vulkan_types.h`.

Reply:
Is there a way to add a namespace for the types defined in `gles_types.h` and `vulkan_types.h`. That will solve the problem.